### ISSUE TITLE
Issues handling bash file descriptors fed to a while loop

### ISSUE THREAD

**Issue by sbhorvatic** [koalaman/shellcheck#2529]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1009, SC1073, SC1056, SC1072
- My shellcheck version (`shellcheck --version` or "online"): 
ShellCheck - shell script analysis tool
version: 0.3.8
license: GNU General Public License, version 3
website: http://www.shellcheck.net

- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#! /bin/bash

function test_fd() {
    local file="${1}"
    while IFS= read -r -u "$searchHits" line
    do
        echo "${line}"
    done {searchHits}< <(grep -n "test_input" "$file")
}

```

#### Here's what shellcheck currently says:
function test_fd() {
^-- SC1009: The mentioned parser error was in this function.
                               ^-- SC1073: Couldn't parse this brace group.
done {searchHits}< <(grep -n "${test_input}" "$file")
         ^-- SC1056: Expected a '}'. If you have one, try a ; or \n in front of it.
         ^-- SC1072: Missing '}'. Fix any mentioned problems and try again.

#### Here's what I wanted or expected to see:
Clean

**Comment by koalaman** [comment]
Fixed in 0f00de80fd seven years ago. Please consider upgrading to a more recent version, 0.3.8 is ancient.

**Comment by sbhorvatic** [comment]
This version was installed with a yum package. Is the package no longer maintained?
